classes/DepositPackage.inc.php

```diff
@@ -598,68 +606,77 @@ public function updateDepositStatus() {
 					'processingState' => $processingState)),
 				SCHEDULED_TASK_MESSAGE_TYPE_NOTICE);
 
+		// Clear previous error messages
 		$this->_deposit->setExportDepositError(null);
+		$this->_deposit->setStagingState($processingState ?: null);
+		// Handle the local state
 		switch ($processingState) {
 			case 'depositedByJournal':
-				$this->_deposit->setTransferredStatus(true);
+			case 'harvest-error':
+				$this->_deposit->setStatus(PLN_PLUGIN_DEPOSIT_STATUS_PACKAGED | PLN_PLUGIN_DEPOSIT_STATUS_TRANSFERRED);
 				break;
 			case 'harvested':
-			case 'xml-validated':
 			case 'payload-validated':
+			case 'bag-validated':
+			case 'xml-validated':
 			case 'virus-checked':
-				$this->_deposit->setReceivedStatus(true);
+			case 'payload-error':
+			case 'bag-error':
+			case 'xml-error':
+			case 'virus-error':
+				$this->_deposit->setStatus(PLN_PLUGIN_DEPOSIT_STATUS_PACKAGED | PLN_PLUGIN_DEPOSIT_STATUS_TRANSFERRED | PLN_PLUGIN_DEPOSIT_STATUS_RECEIVED);
 				break;
-			case 'bag-validated':
 			case 'reserialized':
-				$this->_deposit->setValidatedStatus(true);
-				break;
-			case 'deposited':
-				$this->_deposit->setSentStatus(true);
-				break;
 			case 'hold':
-			case 'harvest-error':
-			case 'deposit-error':
 			case 'reserialize-error':
-			case 'virus-error':
-			case 'xml-error':
-			case 'payload-error':
-			case 'bag-error':
+			case 'deposit-error':
+				$this->_deposit->setStatus(PLN_PLUGIN_DEPOSIT_STATUS_PACKAGED | PLN_PLUGIN_DEPOSIT_STATUS_TRANSFERRED | PLN_PLUGIN_DEPOSIT_STATUS_RECEIVED | PLN_PLUGIN_DEPOSIT_STATUS_VALIDATED);
+				break;
+			case 'deposited':
 			case 'status-error':
-				$this->_deposit->setExportDepositError(__('plugins.generic.pln.status.error.' . $processingState));
+				$this->_deposit->setStatus(PLN_PLUGIN_DEPOSIT_STATUS_PACKAGED | PLN_PLUGIN_DEPOSIT_STATUS_TRANSFERRED | PLN_PLUGIN_DEPOSIT_STATUS_RECEIVED | PLN_PLUGIN_DEPOSIT_STATUS_VALIDATED | PLN_PLUGIN_DEPOSIT_STATUS_SENT);
 				break;
 			default:
 				$this->_deposit->setExportDepositError('Unknown processing state ' . $processingState);
 				$this->_logMessage('Deposit ' . $this->_deposit->getId() . ' has unknown processing state ' . $processingState);
 				break;
 		}
 
+		// The deposit file can be dropped once it's received by the PKP PN
+		if ($this->_deposit->getReceivedStatus()) {
+			$this->remove();
```

Comment:
It's not needed to keep the deposit locally once it was received by the staging server.